mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/query/nativequery/MongoDBNativeQueryPaginationTest.java

```diff
@@ -0,0 +1,202 @@
+			delete( session, liberty );
+			delete( session, vita );
+		} );
+	}
+
+	private void delete(Session session, OscarWildePoem poem) {
+		Object entity = session.get( OscarWildePoem.class, poem.getId() );
+		if ( entity != null ) {
+			session.delete( entity );
+		}
+	}
+
+	@Test
+	public void test_find_page1_10() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_FIND )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 0 )
+				.setMaxResults( 10 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 10 );
+		} );
+	}
+
+	@Test
+	public void test_aggregation_match_page1_10() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_AGGREGATE_MATCH )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 0 )
+				.setMaxResults( 10 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 10 );
+		} );
+	}
+
+	@Test
+	public void test_find_page2_5() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_FIND )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 5 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+		} );
+	}
+
+	@Test
+	public void test_aggregation_match_page2_5() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_AGGREGATE_MATCH )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 5 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+		} );
+	}
+
+	@Test
+	public void test_find_page1_page2_5() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_FIND )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 0 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+
+			result = session.createNativeQuery( QUERY_FIND )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 5 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+		} );
+	}
+
+	@Test
+	public void test_aggregation_match_page1_page2_5() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_AGGREGATE_MATCH )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 0 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+
+			result = session.createNativeQuery( QUERY_AGGREGATE_MATCH )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 5 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+		} );
+	}
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[] { OscarWildePoem.class };
+	}
+
+	public void inTransaction(Consumer<Session> consumer) {
```

Comment:
Nice

Reply:
I've learned it from @dreab8 :)

Reply:
Maybe you can put this method in core so that we can reuse it somewhere else? It doesn't seem specific to this class/dialect

Reply:
move to OgmTestCase base class?

Reply:
I was thinking about  a static method in some utility class. Something like JPAUtils maybe. So that it can be used even  in those test not extending OgmTestCase